test/unit/spec/EnrollUser_spec.js

```diff
@@ -0,0 +1,157 @@
+    var setNextResponse = Util.mockAjax();
+    var baseUrl = 'https://example.okta.com';
+    var logoUrl = 'https://logo.com';
+    var authClient = new OktaAuth({ url: baseUrl, transformErrorXHR: LoginUtil.transformErrorXHR });
+    var router = new Router(_.extend({
+      el: $sandbox,
+      baseUrl: baseUrl,
+      features: { consent: true },
+      logo: logoUrl,
+      authClient: authClient,
+      globalSuccessFn: successSpy
+    }, settings));
+    Util.registerRouter(router);
+    Util.mockRouterNavigate(router);
+    Util.mockJqueryCss();
+
+    router.refreshAuthState('dummy-token');
+    settings = {
+      router: router,
+      successSpy: successSpy,
+      form: new EnrollUserForm($sandbox),
+      ac: authClient,
+      setNextResponse: setNextResponse
+    };
+    if (resUnauthenticatedIdx) {
+      setNextResponse(resUnauthenticatedIdx);
+      return Expect.waitForPrimaryAuth(settings);
+    } else {
+      setNextResponse(resProfileRequiredUpdate);
+      return Expect.waitForEnrollUser(settings);
+    }
+  }
+
+  Expect.describe('Enroll User Form', function () {
+    itp('has the correct title on the enroll form', function () {
+      return setup().then(function (test) {
+        expect(test.form.formTitle().text()).toContain('Create Account');
+      });
+    });
+    itp('has the correct title on the register button', function () {
+      return setup().then(function (test) {
+        expect(test.form.formButton()[0].value).toEqual('Register');
+      });
+    });
+    itp('does not allow empty form submit', function () {
+      return setup().then(function (test) {
+        test.form.formButton().click();
+        expect(test.form.errorMessage()).toEqual('We found some errors. Please review the form and make corrections.');
+      });
+    });
+    itp('renders the right fields based on API response', function () {
+      return setup().then(function (test) {
+        expect(test.form.formInputs('streetAddress').length).toEqual(1);
+        expect(test.form.formInputs('streetAddress').find('input').attr('placeholder')).toEqual('enter streetAddress *'); 
+        expect(test.form.formInputs('streetAddress').hasClass('okta-form-input-field input-fix')).toBe(true);
+
+        expect(test.form.formInputs('employeeId').length).toEqual(1);
+        expect(test.form.formInputs('employeeId').find('input').attr('placeholder')).toEqual('enter employeeId *'); 
+        expect(test.form.formInputs('employeeId').hasClass('okta-form-input-field input-fix')).toBe(true);
+      });
+    });
+
+    itp('makes call to enroll if isEnrollWithLoginIntent is true and then renders the right fields based on API response', function () {
+      return setup(null, resUnauthenticatedIdx).then(function (test) {
+        test.setNextResponse(resProfileRequiredNew);
+        test.form.$('.registration-link').click();
+        return Expect.waitForEnrollUser(test);
+      })
+        .then(function (test) {
```

Comment:
spacing is off here. Other places that you use `.then` it has the same indentation as the previous line (see lines 100 or 110)

Reply:
i moved the `.then` to the same line

```       return setup(null, resUnauthenticatedIdx).then(function (test) {
        test.setNextResponse(resProfileRequiredNew);
        test.form.$('.registration-link').click();
        return Expect.waitForEnrollUser(test);
      }).then(function (test) {
        $.ajax.calls.reset();
        test.setNextResponse(resSuccess);
        expect(test.form.formInputs('streetAddress').length).toEqual(1);
        expect(test.form.formInputs('streetAddress').hasClass('okta-form-input-field input-fix')).toBe(true);
        expect(test.form.formInputs('employeeId').length).toEqual(1);
        expect(test.form.formInputs('employeeId').hasClass('okta-form-input-field input-fix')).toBe(true);
        var model = test.router.controller.model;
        model.set('streetAddress', 'street address');
        model.set('employeeId', '1234');
        model.set('createNewAccount', true);
        model.save();
        return Expect.waitForEnrollUser(test);
      }).then(function () {
        expect($.ajax.calls.count()).toBe(1);
        Expect.isJsonPost($.ajax.calls.argsFor(0), {
          url: 'https://foo.okta.com/api/v1/authn/enroll',
          data: {
            'registration': {
              'createNewAccount': true,
              'profile': {
                'streetAddress': 'street address',
                'employeeId': '1234'
              }
            },
            'stateToken': '01nDL4wRHu-dLvUHUj1QCA9r5P1n5dw6WJ_voGPFWB'
          }
        });
      }); ```